src/deepsparse/server/server.py

```diff
@@ -242,8 +245,9 @@ def _add_pipeline_endpoint(
 
     def _predict(request: pipeline.input_schema):
         pipeline_outputs = pipeline(request)
-        if pipeline.logger:
-            log_resource_utilization(pipeline)
+        server_logger = pipeline.logger
```

Comment:
callback -> attach property to server_logger, lambda function on the side in the callback we know what to log.